driver/src/main/scala/core/nodeset/ChannelFactory.scala

```diff
@@ -71,7 +72,12 @@ private[reactivemongo] final class ChannelFactory(
               s"Connection to ${host}:${port} refused for channel #${chanId}",
               op.cause)
 
-            receiver ! ChannelDisconnected(chanId)
+            op.channel().eventLoop().schedule(new Runnable {
```

Comment:
I don't see the purpose, moreover scheduled/async task should rather go through Akka.

Reply:
The purpose of this is to **not immediately** retry. Current implementation retries immediately when a connection is refused, fails very quickly again, retries again and does this fail-retry cycle **hundreds or thousands times per second** for every channel. There is no reason to retry like this. This impacts the performance of the app since `0.15.1`. CPU is maxed out.  I have provided steps to reproduce it in the referenced mailing list.

`ChannelFactory` currently has no access to the `ActorSystem` or Akka `Scheduler` to schedule a task. If you want I can make a series of changes to constructors of `ChannelFactory`, `StandardDBSystem`, `LegacyDBSystem` etc and pass the `ActorSystem` from within `reactivemongo.api.Driver` or if we don't want to pass the `ActorSystem` I can pass an instance of e.g. `trait Scheduler { def scheduleOnce(delay: FiniteDuration)(block => Unit): Unit }` that I can instantiate using the `ActorSystem` from `reactivemongo.api.Driver`. 

Taking advantage of netty's event loop and scheduler seems simpler in this case and achieves the same purpose but I don't mind scheduling with Akka if you insist at the cost of more changes. Please let me know.

Reply:
hi @cchantep thanks for taking a look and improving this. I've tried your PR #787 and it works well for me so I will close my PR in favour of yours now.